api/v1beta2/awsmachinetemplate_webhook.go

```diff
@@ -224,11 +224,6 @@ func (r *AWSMachineTemplateWebhook) ValidateUpdate(ctx context.Context, oldRaw r
 
 	var allErrs field.ErrorList
 
-	// Allow setting of cloudInit.secureSecretsBackend to "secrets-manager" only to handle v1beta2 upgrade
```

Comment:
This has to be reverted, as upgrade E2E tests would fail

Reply:
`make generate` removes the changes in the manifest.yaml, disabling the validating webhook, which also seems incorrect.

Reply:
Without `make generate` being successful, it wont work.
Not sure why its getting deleted with generate target. 
I will check this, meanwhile tagging @Skarlso @richardcase if they know about it.

Reply:
If you look at the other *webhook.go files, you would notice that only `AWSMachineTemplate` webhook is defined as customvalidator with kubebuilder tags. We should make CustomValidator as Validator, something like:

```go
func (r *AWSMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
	return ctrl.NewWebhookManagedBy(mgr).
		WithValidator(r).
		Complete()
}

// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta2-awsmachinetemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=awsmachinetemplates,versions=v1beta2,name=validation.awsmachinetemplate.infrastructure.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
var _ webhook.Validator = &AWSMachineTemplate{}

```
But this is just a difference I observed which could be the reason its missed out of manifest. Could you give it a try?

Reply:
`AWSMachineTemplate` doesn't implement the Validator interface though, so that won't work either unless we rewrite it. It uses a `CustomValidator` which I am now suspecting is the cause of all this.

Reply:
Yeah but we can implement the missing interfaces, isn't it? Wdyt? I think that will maintain consistency and also would solve our problem.

Reply:
I will try swapping to a regular Validator, but I lack context on why this used a CustomValidator. I don't know if there was a reason behind the difference here.

Reply:
Yes exactly, I think this never worked coz its not there in manifest.
Maybe even I lack the context on why CustomValidator is used from the beginning, but it should not matter I guess. We anyways have to make it work as this is a bug and we should have the validation for AWSMachineTemplate.

cc @richardcase if he knows motivation behind the customvalidator.

Reply:
Refactoring to use the `webhook.Validator` interface fixed the kubebuilder manifest generation. I still do not know how to address the test failures, given that the test descriptions as I understand them disagree with the code.

Reply:
I totally missed the tests part, I will take a look. Thanks for making the changes @cnmcavoy

Reply:
But one thing I am not sure of, that if this is breaking change, then we might have to hold onto this PR till next api version change.

Reply:
The API break is around the removal of the `AWSMachineTemplateWebhook` struct. How strict are the api-breaking rules? I'd be shocked if removing that type mattered to any API consumer.

However, if we must preserve 1:1 compatibility,  we could leave that type and wire it into the validator for backwards compat.

Reply:
I believe I satisfied the api-break by re-adding the removed types and deprecating them all. That should remove that blocker.

Reply:
It is used internally in CAPA but there might be consumers who would directly use APIs in their projects, so it would be better if we could leave the type and wire it for backward compatibility.

Reply:
@richardcase @Skarlso would also need your opinion on this.

Reply:
I am sorry to make this ping pong but looks like CustomValidator has been introduced for SSA feature. But I am not sure how to make this correct in the manifest.
I will give it another look once again before you make any new changes.

Reply:
See [this](https://cluster-api.sigs.k8s.io/developer/providers/webhooks.html#validating-webhooks) for details.

Reply:
So that would mean we just have to debug our generate target, but rest all code should remain same.

Reply:
I figured out the issue with `make generate`. The kubebuilder comment needs a line of blank whitespace following it. Inserting a blank line after L46 of `awsmachinetemplates.go` causes the manifest to be added to the yaml.

I'm not sure why the whitespace matters to kubebuilder though.

Reply:
I reverted this back to where it was before this thread sent us on the wrong path, but with the addition of the extra blank line.

The test failures still need to be resolved.

Reply:
Based on the comment and the previous git history, I see that I might have changed `SecretBackendSecretsManager` to `SecretBackendSSMParameterStore` unintentionally. Could you change it back to SecretBackendSecretsManager in the if condition and see if it works

Reply:
Now that I think about it again, I think you were right initially that this check doesn't affect us anymore. Let me dig out the original issue for this check.

Reply:
Thanks for digging up that history @Ankitasw, that explains much. I updated the PR to "fix" the behavior as you documented, then added a followup commit to remove the defaulting behavior and tests.

Assuming the e2e passes, this should be ready to review / merge.